test/plugin.js

```diff
@@ -207,6 +218,60 @@ describe("nemo-drivex @plugin@", function () {
     });
   });
   /**
+   * waitForElementNotVisible Wait for timeout milliseconds for the WebElement to be NOT visible
+   * @param locator {LocatorJSON}
+   * @param timeout {Number}
+   * @param msg {String} optional message for any error messages
+   * @returns {Promise} resolves to true or false
+   */
+  describe("@waitForElementNotVisible@ method", function () {
+    it("should not be visible", function () {
+      assert.ok(nemo.drivex.waitForElementNotVisible);
+    });
+    it("should return true when element is not visible", function (done) {
+      nemo.driver.get('https://warm-river-3624.herokuapp.com/waits');
+      util.waitForJSReady(nemo);
+      nemo.drivex.waitForElementNotVisible({
+        'locator': 'outy',
+        'type': 'id'
+      }, 6000, "found outy div visible").then(function (notVisible) {
+        assert.equal(notVisible, true);
+        done();
+      }, util.doneSuccess(done));
+    });
+    it("should reject promise when element is visible", function (done) {
+      nemo.driver.get('https://warm-river-3624.herokuapp.com/waits');
+      util.waitForJSReady(nemo);
+      nemo.drivex.find({locator: '#wrapper > form > input', type: 'css'}).click();
+      nemo.driver.sleep(5000); // Waiting for element to be visible
+      nemo.drivex.waitForElementNotVisible({
+        'locator': 'outy',
+        'type': 'id'
+      }, 6000, "found outy tag").then(function (notVisible) {
+        assert.equal(notVisible, false);
+        done();
+      }, util.doneSuccess(done));
+    });
+    it("should reject promise when element is not present, but only after the appropriate timeout", function (done) {
+      var startMS;
+      nemo.driver.get('https://warm-river-3624.herokuapp.com/waits');
+      util.waitForJSReady(nemo).then(function() {
+        startMS = Date.now();
+      });
+      nemo.drivex.waitForElementNotVisible({
+        'locator': 'idontexist',
+        'type': 'id'
+      }, 3000, "couldn't find idontexist as not visible").then(function (elt) {
+        done(new Error('shouldnt have got here'));
+      }, function (err) {
+        var foundMS = Date.now() - startMS;
+        assert(err);
+        assert.ok(foundMS > 3000);
+        done();
+      });
+    });
+  });
```

Comment:
We should have an additional test case which tests that a visible element becomes notVisible. It could be achieved by:
1. use `driver.executeScript` to execute a `setTimeout(hideScript, 2000)` in the DOM of the app under test where the `hideScript` will hide a visible dom element 2 seconds hence
2. test execute a waitForElementNotVisible on that dom element
3. check both that waitForElementNotVisible returns true AND that it returns true in a timeframe consistent with the setTimeout value in step 1